flow-decomposition/src/test/java/com/powsybl/flow_decomposition/FlowDecompositionTests.java

```diff
@@ -196,6 +196,15 @@ void testFlowDecompositionOnHvdcNetwork() {
         assertTrue(flowDecompositionResults.getZoneSet().contains(Country.NL));
     }
 
+    @Test
+    void testConnectedComponentModeChangesFromAllToMain() {
+        Network network = TestUtils.getMicroGridNetworkWithBusBarSectionOnly();
+        LoadFlowParameters loadFlowParameters = new LoadFlowParameters().setConnectedComponentMode(LoadFlowParameters.ConnectedComponentMode.ALL);
+        FlowDecompositionComputer flowDecompositionComputer = new FlowDecompositionComputer(new FlowDecompositionParameters(), loadFlowParameters);
+        flowDecompositionComputer.run(new XnecProviderAllBranches(), network);
```

Comment:
I do not think you need to run the flow decomposition here

Reply:
Indeed!